compiler/rustc_passes/src/liveness.rs

```diff
@@ -1686,6 +1710,38 @@ impl<'tcx> Liveness<'_, 'tcx> {
         }
     }
 
+    fn has_added_lit_match_name_span(
+        &self,
+        name: &str,
+        opt_body: Option<&hir::Body<'_>>,
+        err: &mut rustc_errors::DiagnosticBuilder<'_, ()>,
+    ) -> bool {
+        let mut has_litstring = false;
+        if let Some(opt_body) = opt_body {
+            let mut visitor = CollectLitsVisitor { lit_exprs: vec![] };
+            intravisit::walk_body(&mut visitor, opt_body);
+            for lit_expr in visitor.lit_exprs {
+                if let hir::ExprKind::Lit(litx) = &lit_expr.kind {
```

Comment:
and 

```suggestion
                let hir::ExprKind::Lit(litx) = &lit_expr.kind else { continue; };
```

Reply:
I meant to literally use

```rust
let hir::ExprKind::Lit(litx) = &lit_expr.kind else { continue };
```

not `if let`, but `let else` syntax

Reply:
this would remove nesting, as you then have access to `litx` in the next line.

Reply:
fixed and code likes cleaner. thanks